File: Application/app/src/main/java/com/example/code/SegmentSeekBar.java
package com.example.code;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.Color;
import android.graphics.Paint;
import android.graphics.Rect;
import android.util.AttributeSet;

import androidx.appcompat.widget.AppCompatSeekBar;

public class SegmentSeekBar extends AppCompatSeekBar {

    private Paint mRulerPaint;
    private int mRulerCount = 4;
    private int mRulerWidth = 2;
    private int mRulerColor = Color.WHITE;
    private boolean isShowTopOfThumb = false;


    public SegmentSeekBar(Context context) {
        super(context);
        init();
    }

    public SegmentSeekBar(Context context, AttributeSet attrs) {
        super(context, attrs);
        init();
    }

    public SegmentSeekBar(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init();
    }

    private void init() {
        mRulerPaint = new Paint();
        mRulerPaint.setColor(mRulerColor);
        mRulerPaint.setAntiAlias(true);
        setSplitTrack(false);
    }

    @Override
    protected synchronized void onDraw(Canvas canvas) {
        super.onDraw(canvas);

        if (getWidth() <= 0 || mRulerCount <= 0) {
            return;
        }

        int length = (getWidth() - getPaddingLeft() - getPaddingRight() - mRulerCount * mRulerWidth) / (mRulerCount + 1);

        int rulerTop = getHeight() / 2 - getMinimumHeight() / 2;
        int rulerBottom = rulerTop + getMinimumHeight();

        Rect thumbRect = null;
        if (getThumb() != null) {
            thumbRect = getThumb().getBounds();
        }

        for (int i = 1; i <= mRulerCount; i++) {
            int rulerLeft = i * length + getPaddingLeft();
            int rulerRight = rulerLeft + mRulerWidth;

            if (!isShowTopOfThumb && thumbRect != null && rulerLeft - getPaddingLeft() > thumbRect.left && rulerRight - getPaddingLeft() < thumbRect.right) {
                continue;
            }

            canvas.drawRect(rulerLeft, rulerTop, rulerRight, rulerBottom, mRulerPaint);
        }
    }

    public void setRulerCount(int mRulerCount) {
        this.mRulerCount = mRulerCount;
        requestLayout();
    }

    public void setRulerWidth(int mRulerWidth) {
        this.mRulerWidth = mRulerWidth;
        requestLayout();
    }

    public void setRulerColor(int mRulerColor) {
        this.mRulerColor = mRulerColor;
        if (mRulerPaint != null) {
            mRulerPaint.setColor(mRulerColor);
            requestLayout();
        }
    }

    public void setShowTopOfThumb(boolean isShowTopOfThumb) {
        this.isShowTopOfThumb = isShowTopOfThumb;
        requestLayout();
    }
}

File: Application/app/src/main/java/com/example/code/SegmentSeekBar.kt
package com.example.code

import android.content.Context
import android.graphics.Canvas
import android.graphics.Color
import android.graphics.Paint
import android.graphics.Rect
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatSeekBar
import kotlin.jvm.Synchronized

class SegmentSeekBar : AppCompatSeekBar {
    private var segmentDividerPaint: Paint? = null
    private var segmentsCount = 0
    private var segmentDividerWidth = 3
    private var segmentDividerColor = Color.WHITE
    private var isShowTopOfThumb = false

    constructor(context: Context?) : super(context!!) { init() }

    constructor(context: Context?, attrs: AttributeSet?) : super(context!!, attrs) { init() }

    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int) : super(
        context!!, attrs, defStyleAttr
    ) { init() }

    private fun init() {
        segmentDividerPaint = Paint()
        segmentDividerPaint?.color = segmentDividerColor
        segmentDividerPaint?.isAntiAlias = true
        splitTrack = false
    }

    @Synchronized
    override fun onDraw(canvas: Canvas) {
        super.onDraw(canvas)

        if (width <= 0 || segmentsCount <= 0) { return }

        val length =
            (width - paddingLeft - paddingRight - segmentsCount * segmentDividerWidth) / (segmentsCount + 1)
        val rulerTop = height / 1.2 - minimumHeight / 2
        val rulerBottom = rulerTop/ 1.4 + minimumHeight /2.9

        var thumbRect: Rect? = null
        if (thumb != null) {
            thumbRect = thumb.bounds
        }

        for (i in 1..segmentsCount) {
            val rulerLeft = i * length + paddingLeft
            val rulerRight = rulerLeft + segmentDividerWidth
            if (!isShowTopOfThumb &&
                thumbRect != null &&
                rulerLeft - paddingLeft > thumbRect.left &&
                rulerRight - paddingLeft < thumbRect.right) {

                continue

            }
            canvas.drawRect(rulerLeft.toFloat(), rulerTop.toFloat(), rulerRight.toFloat(),
                rulerBottom.toFloat(), segmentDividerPaint!!
            )
        }
    }

    fun setMplSegmentStyle(segmentsList: List<Segment>,
                           segmentDividerColor:Int,segmentDividerWidth: Int
    ) {
        setSegmentsCount(segmentsList.size)
        setSegmentDividerColor(segmentDividerColor)
        setSegmentDividerWidth(segmentDividerWidth)
        requestLayout()
    }

    private fun setSegmentsCount(mRulerCount: Int) { this.segmentsCount = mRulerCount }

    private fun setSegmentDividerColor(segmentDividerColor: Int) {
        this.segmentDividerColor = segmentDividerColor
        if (segmentDividerPaint != null) {
            segmentDividerPaint?.color = segmentDividerColor
        }
    }

    private fun setSegmentDividerWidth(segmentDividerWidth: Int) {
        this.segmentDividerWidth = segmentDividerWidth
    }


}